### PR TITLE
Add LatestHead relayer method to gRPC client and server implementations

### PR DESCRIPTION
**Pull request by kidambisrinivas**
## Updates

- Add LatestHead relayer function to gRPC client-server interfaces and implementations
  - Add LatestHead implementation to relayerset server and clients

## Build

```
$ go generate ./...
``` 

## Testing

```
$ go test -count=1 -v ./pkg/loop/internal/relayerset
=== RUN   Test_RelayerSet
2024-09-13T11:08:35.420+0100 [DEBUG] accepting initial connection: addr=/var/folders/1b/byq7w64n1x7gmp5cp9__vp_r0000gn/T/plugin3130643913
2024-09-13T11:08:35.420+0100 [DEBUG] stdio EOF, exiting copy loop
2024-09-13T11:08:35.420+0100 [DEBUG] stdio EOF, exiting copy loop
2024-09-13T11:08:35.421+0100 [DEBUG] making new client mux initial connection: addr=/var/folders/1b/byq7w64n1x7gmp5cp9__vp_r0000gn/T/plugin3130643913
2024-09-13T11:08:35.421+0100 [DEBUG] initial server connection accepted: addr=/var/folders/1b/byq7w64n1x7gmp5cp9__vp_r0000gn/T/plugin3130643913
2024-09-13T11:08:35.421+0100 [DEBUG] client muxer connected: addr=/var/folders/1b/byq7w64n1x7gmp5cp9__vp_r0000gn/T/plugin3130643913
2024-09-13T11:08:35.421+0100 [DEBUG] sending conn to default listener
2024-09-13T11:08:35.424+0100 [DEBUG] sending conn to default listener
2024-09-13T11:08:35.424+0100 [DEBUG] stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
--- PASS: Test_RelayerSet (0.00s)
PASS
ok      github.com/smartcontractkit/chainlink-common/pkg/loop/internal/relayerset       0.398s
```

### CODE REVIEW

File: pkg/loop/internal/relayerset/client.go [hunk 1]
```diff
@@ -100,6 +100,21 @@ func (k *Client) RelayerName(ctx context.Context, relayID types.RelayID) (string
...
 }
 
+func (k *Client) RelayerLatestHead(ctx context.Context, relayID types.RelayID) (types.Head, error) {
+	req := &relayerset.LatestHeadRequest{
+		RelayerId: &relayerset.RelayerId{ChainId: relayID.ChainID, Network: relayID.Network},
+	}
+	resp, err := k.relayerSetClient.RelayerLatestHead(ctx, req)
+	if err != nil {
+		return types.Head{}, fmt.Errorf("error getting new chain writer: %w", err)
+	}
```
Comment (ilija42) [inline pkg/loop/internal/relayerset/client.go:110]: This error message doesn't make sense

File: pkg/loop/internal/pb/relayerset/relayerset.proto
Comment (jmank88) [inline pkg/loop/internal/pb/relayerset/relayerset.proto:1]: It looks like we have been mixing field name styles:
- `legacy_url`
- `transmitterID`
- `relayerId`
- `plugin_provider_id`

**Review by jmank88 — APPROVED** [review]

File: pkg/loop/internal/pb/relayerset/relayerset.proto
Comment (cedric-cordenier) [reply pkg/loop/internal/pb/relayerset/relayerset.proto:1]: @kidambisrinivas Do you mind following up on Jordan's comment, either in this PR or in the next?

**Review by cedric-cordenier — APPROVED** [review]

File: pkg/loop/internal/pb/relayerset/relayerset.proto
Comment (kidambisrinivas) [reply pkg/loop/internal/pb/relayerset/relayerset.proto:1]: In gRPC code apparently, this style is the convention: relayerId

File: pkg/loop/internal/relayerset/client.go (hunk 1 above)
Comment (kidambisrinivas) [reply pkg/loop/internal/relayerset/client.go]: Fixed this

**Review by jmank88 — APPROVED** [review]

**Review by cedric-cordenier — APPROVED** [review]

File: pkg/loop/internal/pb/relayerset/relayerset.proto
Comment (kidambisrinivas) [reply pkg/loop/internal/pb/relayerset/relayerset.proto:1]: Yeah, it'll affected generated go code and would likely be disruptive. I could handle this as a separate PR, as the naming styles are prior to this PR and would be a larger change in general

Comment (jmank88) [reply pkg/loop/internal/pb/relayerset/relayerset.proto:1]: > In gRPC code apparently, this style is the convention: relayerId

Do you have a reference for this?

Comment (jmank88) [reply pkg/loop/internal/pb/relayerset/relayerset.proto:1]: eh I guess I see `Id` here: https://protobuf.dev/getting-started/gotutorial/#protobuf-api

**Pull request merged**